linked_list/linked_list.py

```diff
@@ -89,29 +170,87 @@ def visit(self):
     # Time Complexity: ?
     # Space Complexity: ?
     def reverse(self):
-        pass
+        if self.head == None:
+            return 
+        elif self.head.next == None:
+            return
+        else:
+            current = self.head
+            reversed_link = None
+            while current.next != None:
+                if not reversed_link:
+                    next_node = current.next
+                    current.next = None
+                    reversed_link = current
+                    current = next_node
+                else:

```

Comment:
hmm I don't think we need this first if statement if we change the while loop to `while current != None`. I think starting `reversed_link` as `None` like you did and then going into the `else` statement will work